src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
@janvorli I made a draft (I did not update the PR yet) for your suggestion but it does not work with nested exception handling like:

```cs
main {
  try {
    try {
      Console.WriteLine("try, first throw");
      throw new Exception("a");
    } catch (Exception) {
      Console.WriteLine("first catch, second throw");
      throw new Exception("b");
    }
  } catch (Exception) {
      Console.WriteLine("second catch");
    return 2;
  }
  return 0;
}
```

And the error message is:
```
try, first throw
first catch, second throw
terminate called after throwing an instance of 'PAL_SEHException'
```

It crashes when `PAL_ThrowExceptionFromContext()` throws C++ exception. IMHO it went wrong when unwinding the helper(CallEHFunclet) frame which is the caller of inner catch funclet, since there is no funclet(or its helper) when it handled first one.

Here is my draft https://github.com/wateret/coreclr/commit/632f2a6e03de2d641dcecaf40811ede26b26bf5b . It would be nice if you have a look. Thanks.